src/blocktree.erl

```diff
@@ -24,80 +31,34 @@
 -type blockid() :: integer() | undefined.
 -type blockmap() :: map().
 -type block() :: map().
--type playernoncemap() :: map().
-
--spec add_new_transaction_to_array(tx(), txarray()) ->
-	{addresult(), txarray()}.
-add_new_transaction_to_array(Transaction, TransactionArray)
-  when is_map(Transaction) ->
-
-    Nonce = maps:get(nonce, Transaction),
-    TA = TransactionArray,
-    ArrSz = array:size(TA),
 
-    if
-	Nonce > ArrSz ->
-	    {ignore_nonce_too_high, TA};
-	Nonce < ArrSz ->
-	    RecordedTransaction = array:get(Nonce, TA),
-	    if
-		RecordedTransaction == Transaction ->
-		    {ignore_duplicate, TA};
-		RecordedTransaction /= Transaction ->
-		    throw("same_nonce_different_transaction")
-	    end;
-	Nonce == ArrSz ->
-	    NewTA = array:set(Nonce, Transaction, TA),
-	    {added, NewTA}
-    end.
-
--spec add_new_transaction_to_map(tx(), playertxmap()) -> {addresult(), playertxmap()}.
-add_new_transaction_to_map(Transaction, TransactionMap)
-  when is_map(Transaction) ->
-
-    Id = maps:get(player_id, Transaction),
-    TM = TransactionMap,
-    Result = maps:find(Id, TM),
-
-    case Result of
-	{ok, TransactionArray} ->
-	    pass;
-	error ->
-	    TransactionArray = array:new()
-    end,
-
-    {Msg, NewTA} = add_new_transaction_to_array(Transaction, TransactionArray),
-    NewTM = maps:put(Id, NewTA, TM),
-    {Msg, NewTM}.
+%% @doc Initializes an empty container
+-spec init() -> treedata().
+init() ->
```

Comment:
how about calling this new_treedata() instead?

Reply:
YA

Reply:
wait no

Reply:
maybe?

Reply:
how about just new()?

blocktree:new()

Reply:
anything but init 😱